ragna/source_storages/_chroma.py

```diff
@@ -100,22 +101,19 @@ def retrieve(
                 max(int(num_tokens * 2 / chunk_size), 100),
                 collection.count(),
             ),
-            include=["distances", "metadatas", "documents"],
+            include=include,
         )
 
         num_results = len(result["ids"][0])
-        result = {
-            key: [None] * num_results if value is None else value[0]  # type: ignore[index]
-            for key, value in result.items()
-        }
+        result = {key: result[key][0] for key in ["ids", *include]}
```

Comment:
Question: Handling the `None` case is no longer relevant?

Reply:
Correct. Anything that could be returned by Chroma, but is not present in the `include` list, results in `None` values. Since we no longer iterate over the full dictionary, we no longer need to care about them.